### PR TITLE
Simplify Chinese translation of Scala Tour: variances.md

### PR DESCRIPTION
**Pull request by realwunan**
Simplify Chinese translation of Scala Tour: Variances

### CODE REVIEW

File: _zh-cn/tour/variances.md [hunk 1]
```diff
@@ -13,3 +13,143 @@ language: zh-cn
...
+另一个可以帮助理型变的例子是来自 Scala 标准库的 `trait Function1[-T, +R]`。 `Function1` 表示具有一个参数的函数，其中第一个类型参数 `T` 表示参数类型，第二个类型参数 `R` 表示返回类型。 `Function1` 在其参数类型上是逆变的，并且在其返回类型上是协变的。 对于这个例子，我们将使用文字符号 `A => B` 来表示 `Function1[A, B]`。
+
+假设前面使用过的类似 `Cat`，`Dog`，`Animal` 的继承关系，加上以下内容：
+
+```tut
+abstract class SmallAnimal extends Animal
+case class Mouse(name: String) extends SmallAnimal
+```
+
+假设我们正在处理接受动物类型的函数，并返回他们的食物类型。 如果我们想要一个 `Cat => SmallAnimal`（因为猫吃小动物），但是给它一个 `Animal => Mouse`，我们的程序仍然可以工作。 直观地看，一个 `Animal => Mouse` 的函数仍然会接受一个 `Cat` 作为参数，因为 `Cat` 即是一个 `Animal`，并且这个函数返回一个 `Mouse`，也是一个 `SmallAnimal`。 既然我们可以安全地，无形地用前者代替后者，我们可以说 `Animal => Mouse` 是 `Cat => SmallAnimal` 的子类型。
```
Comment (dongxuwang) [inline _zh-cn/tour/variances.md]:
```
-无形
+隐式
```

File: _zh-cn/tour/variances.md [hunk 2]
```diff
@@ -13,3 +13,143 @@ language: zh-cn
...
+这个程序的输出如下：
+
+```
+The cat's name is: Boots
+The animal's name is: Boots
+```
+
+### 不变
+
+默认情况下，Scala中的通用类是不变的。 这意味着它们既不是协变的也不是逆变的。 在下例中，类 `Container` 是不变的。 `Container[Cat]` _不是_ `Container[Animal]`，反之亦然。
```
[the remark this answers is not in the document]
Comment (realwunan) [reply _zh-cn/tour/variances.md]: 对，这里疏忽了，谢谢

File: _zh-cn/tour/variances.md [hunk 3]
```diff
@@ -13,3 +13,143 @@ language: zh-cn
...
+```
+val catContainer: Container[Cat] = new Container(Cat("Felix"))
+val animalContainer: Container[Animal] = catContainer
+animalContainer.setValue(Dog("Spot"))
+val cat: Cat = catContainer.getValue // 糟糕，我们最终会将一只狗作为值分配给一只猫
+```
+
+幸运的是，编译器在此之前就会阻止我们。
+
+### 别的例子
```
[the remark this answers is not in the document]
Comment (realwunan) [reply _zh-cn/tour/variances.md]: Thanks

File: _zh-cn/tour/variances.md [hunk 4]
```diff
@@ -13,3 +13,143 @@ language: zh-cn
...
+
+另一个可以帮助理型变的例子是来自 Scala 标准库的 `trait Function1[-T, +R]`。 `Function1` 表示具有一个参数的函数，其中第一个类型参数 `T` 表示参数类型，第二个类型参数 `R` 表示返回类型。 `Function1` 在其参数类型上是逆变的，并且在其返回类型上是协变的。 对于这个例子，我们将使用文字符号 `A => B` 来表示 `Function1[A, B]`。
```
[the remark this answers is not in the document]
Comment (realwunan) [reply _zh-cn/tour/variances.md]: Thanks a lot

File: _zh-cn/tour/variances.md (hunk 1 above)
Comment (realwunan) [reply _zh-cn/tour/variances.md]: Thanks a lot.

**Comment by SethTisue** [comment]
thank you!

**Pull request merged**